### PR TITLE
conmon: Add support for partial/newline log tags

### PR DESCRIPTION
**Pull request by mrunalp**
This is for https://github.com/kubernetes/kubernetes/pull/55922

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

**- What I did**
Added support for partial/newline log tags.

**- How I did it**
Modified conmon to add the log tags. This simplifies log writing logic.

**- How to verify it**
kubectl logs should work fine like before.

**- Description for the changelog**
```changelog
Add support for partial/newline log tags
```

### CODE REVIEW

File: conmon/conmon.c [hunk 1]
```diff
@@ -370,18 +354,21 @@ static int write_k8s_log(int fd, stdpipe_t pipe, const char *buf, ssize_t buflen
 			fd = log_fd;
 		}
 
-		/* Output a newline */
-		if (insert_newline) {
-			if (writev_buffer_append_segment(fd, &bufv, "\n", -1) < 0) {
-				nwarn("failed to write newline to log");
+		/* Output the timestamp */
```
Comment (rhatdan) [inline conmon/conmon.c]: Looks like a formatting issue.

File: conmon/conmon.c [hunk 2]
```diff
@@ -370,18 +354,21 @@ static int write_k8s_log(int fd, stdpipe_t pipe, const char *buf, ssize_t buflen
...
 			}
-		}
 
-		/* Output a timestamp */
-		if (insert_timestamp) {
-			if (writev_buffer_append_segment(fd, &bufv, tsbuf, -1) < 0) {
-				nwarn("failed to write (timestamp, stream) to log");
+		/* Output log tag for partial or newline */
+		if (partial) {
+			if (writev_buffer_append_segment(fd, &bufv, "P ", -1) < 0) {
```
Comment (rhatdan) [inline conmon/conmon.c]: Is something upstream from conmon interpreting this "E " and "P "?

Comment (mrunalp) [reply conmon/conmon.c]: Yeah, the linked kubernetes PR does that. We need to wait for that to be finalized and merged. I just wanted to make sure that we can handle this in conmon.

**Comment by mrunalp** [comment]
Updated.

**Comment by mrunalp** [comment]
Updated tests and execsync parse logic.

File: test/ctr.bats [hunk 3]
```diff
@@ -238,7 +238,7 @@ function teardown() {
 	logpath="$DEFAULT_LOG_PATH/$pod_id/$ctr_id.log"
 	[ -f "$logpath" ]
 	echo "$logpath :: $(cat "$logpath")"
-	grep --binary -P "^[^\n]+ stdout here is some output\x0d$" "$logpath"
+	grep --binary -P "^[^\n]+ stdout F here is some output\x0d$" "$logpath"
```
Comment (runcom) [inline test/ctr.bats]: Could we need some test for the other cases added?

File: contrib/test/integration/build/kubernetes.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 - name: clone kubernetes source repo
   git:
-    repo: "https://github.com/runcom/kubernetes.git"
+    repo: "https://github.com/{{ k8s_github_fork }}/kubernetes.git"
```
Comment (TomSweeneyRedHat) [inline contrib/test/integration/build/kubernetes.yml]: 👍

File: test/ctr.bats (hunk 3 above)
Comment (mrunalp) [reply test/ctr.bats]: Adding a test for partial line.

Comment (mrunalp) [reply test/ctr.bats]: Pushed the test.

**Comment by mrunalp** [comment]
ami build failed in critest validation. Updated commits to update cri-tools to latest to support new log format.

**Comment by runcom** [comment]
needs rebase and tests still failing :cry:

**Comment by mrunalp** [comment]
Rebased and updated.

**Comment by mrunalp** [comment]
https://github.com/vbatts/git-validation/commit/8ebf5d3f1064fff8759bc3dc84f1c3c61ddb33bd is breaking e2e. Fixing.

**Comment by mrunalp** [comment]
@runcom The ami tests passed. This is ready to merge.

**Comment by runcom** [comment]
@mrunalp why do we need another ami? This targets the master branch, we should be able to make tests passing w/o touching the amis :/

**Comment by runcom** [comment]
we can remove git validation from e2e btw, we don't ever need that there.

**Comment by runcom** [comment]
This should fix this https://github.com/kubernetes-incubator/cri-o/pull/1181

**Comment by runcom** [comment]
I actually don't understand how /test ami passes then git-validation import collision while e2e fails......

**Comment by mrunalp** [comment]
The reason ami passes is because it repulls latest git-validation which has lowercase sirupsen. If we just run e2e, then it is repulling git-validation on top of ami cached version that had already pulled upper-case Sirupsen causing the error. Hence, the ami passes :)

**Comment by mrunalp** [comment]
Rebased.

**Pull request merged**